src/pages/[chapterId]/index.tsx

```diff
@@ -6,64 +6,72 @@ import Error from 'next/error';
 import { getChapterVerses } from 'src/api';
 import NextSeoHead from 'src/components/NextSeoHead';
 import QuranReader from 'src/components/QuranReader';
+import { QuranReaderDataType } from 'src/components/QuranReader/types';
 import { getDefaultWordFields } from 'src/utils/api';
 import { getChapterData } from 'src/utils/chapter';
+import { getVerseNavigationUrl } from 'src/utils/navigation';
 import {
   REVALIDATION_PERIOD_ON_ERROR_SECONDS,
   ONE_WEEK_REVALIDATION_PERIOD_SECONDS,
 } from 'src/utils/staticPageGeneration';
-import { isValidChapterId } from 'src/utils/validator';
+import { isValidChapterId, isValidVerseKey } from 'src/utils/validator';
 import { ChapterResponse, VersesResponse } from 'types/ApiResponses';
 
 type ChapterProps = {
   chapterResponse?: ChapterResponse;
   versesResponse?: VersesResponse;
   hasError?: boolean;
+  isChapter?: boolean;
 };
 
 const Chapter: NextPage<ChapterProps> = ({ chapterResponse, versesResponse, hasError }) => {
   if (hasError) {
     return <Error statusCode={500} />;
   }
-
   return (
     <>
       <NextSeoHead
         title={`Surah ${chapterResponse.chapter.nameSimple} - 1-${chapterResponse.chapter.versesCount}`}
       />
-      <QuranReader initialData={versesResponse} id={chapterResponse.chapter.id} />
+      <QuranReader
+        initialData={versesResponse}
+        id={chapterResponse.chapter.id}
+        quranReaderDataType={QuranReaderDataType.Chapter}
+      />
     </>
   );
 };
 
 export const getStaticProps: GetStaticProps = async ({ params, locale }) => {
-  const chapterId = String(params.chapterId);
-  // we need to validate the chapterId first to save calling BE since we haven't set the valid paths inside getStaticPaths to avoid pre-rendering them at build time.
-  if (!isValidChapterId(chapterId)) {
+  const chapterIdOrVerseKey = String(params.chapterId);
+  const isChapter = isValidChapterId(chapterIdOrVerseKey);
+  // if it's not a valid chapter id and it's not a valid verse key, we reject it.
+  if (!isChapter && !isValidVerseKey(chapterIdOrVerseKey)) {
+    return { notFound: true };
+  }
+  // if it's a verseKey, we redirect to the url of the verse
+  if (!isChapter) {
```

Comment:
Does it make sense to do if !isChapter && isVerseKey to be extra safe? 

I can imagine us expanding the scope in the future